### PR TITLE
Intro Tutorial fixes

### PR DESCRIPTION
**Pull request by tpike3**
Fixed the build issues. The read the docs now reflects the corrections in the introductory tutorial.

### CODE REVIEW

**Comment by tpike3** [comment]
Detailed description of fixes: 

1. How do users get the requirements.txt?
FIX: Put in link directly to requirements text for pip install

 2. MoneyModel.py - is the casing correct following coding standards for python files?
FIX: MoneyModel.py references changed to money_model.py

 3. Citation link is broken: Comer2014. URL: https://mesa.readthedocs.io/en/master/tutorials/intro_tutorial.html#comer2014
FIX: Changed link to direct George Mason MARs .pdf of Comer dissertation 

 4. "Every agent is expected to have a step method, which takes a model object as its only argument".
FIX: Reworded sentences to remove model and make description more explicit

 5. In "Running your first model" at the phrase: "You’ll probably see something like the distribution shown below." the plot is actually above this part in the narrative.
FIX: Moved description to above plot

 6. The moore argument to get_neighborhood might be unclear what this does for new users.
FIX: Linked Moore and Von Neumann to wikipedia articles about each approach

 7. defining give_money
FIX: turned other references into other_agent to make more explicit

 8. In "Adding Space", after the line "Now, putting that all together should look like this:"
FIX: made order the same

 9. Under "Batch Run" - from mesa.batchrunner import BatchRunner gets imported, but not used in the next code block.
FIX: moved batchrunner import cell to above batchrunner import

 10. In the batch runner example, we do not import compute_gini.
FIX: Added gini method to make more explicit

File: docs/tutorials/intro_tutorial.rst [hunk 1]
```diff
@@ -10,22 +9,21 @@ Tutorial Description
...
+money. Next, we add *space* to allow agents move. Then, we’ll cover two
+of Mesa’s analytic tools: the *data collector* and *batch runner*. After
+that, we’ll add an *interactive visualization* which lets us watch the
 model as it runs. Finally, we go over how to write your own
 visualization module, for users who are comfortable with JavaScript.
 
 You can also find all the code this tutorial describes in the
-`examples/boltzmann_wealth_model/ <https://github.com/projectmesa/mesa/tree/master/examples/boltzmann_wealth_model>`__ directory of the Mesa
-repository.
+**examples/boltzmann_wealth_model** directory of the Mesa repository.
```
Comment (Corvince) [inline docs/tutorials/intro_tutorial.rst:26]: Why remove the link?

File: docs/tutorials/intro_tutorial.rst [hunk 2]
```diff
@@ -384,213 +348,208 @@ those on the left edge, and the top to the bottom. This prevents some
...
-        def move(self):
-            possible_steps = self.model.grid.get_neighborhood(
-                self.pos,
-                moore=True,
-                include_center=False
-            )
-            new_position = self.random.choice(possible_steps)
-            self.model.grid.move_agent(self, new_position)
+   class MoneyAgent(Agent):
+      #...
```
Comment (Corvince) [inline docs/tutorials/intro_tutorial.rst:428]: indentation: missing one space

File: docs/tutorials/intro_tutorial.rst [hunk 3]
```diff
@@ -384,213 +348,208 @@ those on the left edge, and the top to the bottom. This prevents some
...
+First let’s handle movement, and have the agents move to a neighboring
+cell. The grid object provides a ``move_agent`` method, which like you’d
 imagine, moves an agent to a given cell. That still leaves us to get the
 possible neighboring cells to move to. There are a couple ways to do
 this. One is to use the current coordinates, and loop over all
 coordinates +/- 1 away from it. For example:
 
 .. code:: python
 
-    neighbors = []
```
Comment (Corvince) [inline docs/tutorials/intro_tutorial.rst:444]: Here and below: Why change the indentation from 4 spaces to 3 spaces? Any special meaning to that?

File: docs/tutorials/intro_tutorial.rst (hunk 1 above)
Comment (tpike3) [reply docs/tutorials/intro_tutorial.rst:26]: Must have done it accidentally...no issues keeping the link in

File: docs/tutorials/intro_tutorial.rst (hunk 2 above)
Comment (tpike3) [reply docs/tutorials/intro_tutorial.rst:428]: Sorry a little new to this, do you want me to change this? I can add the space

File: docs/tutorials/intro_tutorial.rst (hunk 3 above)
Comment (tpike3) [reply docs/tutorials/intro_tutorial.rst:444]: No special meaning, most likely inadvertent

**Comment by tpike3** [comment]
Regarding the .rst, ipynb etc. I was following the existing file structure. My understanding is the tutorial is written in .ipynb, saved as an .rst for read the docs and then produced for the read the docs using sphinx/readthedocs. I wasn't exactly clear on that and trying to get it so the .rst reflected the changes in the .ipynb caused a lost of commits.

**Comment by Corvince** [comment]
Hmm I guess it is kind of nice to have an interactive jupyter notebook file for the tutorials, but is this actually documented somewhere? Otherwise we should really provide a link to that one so people know they can run the tutorial locally and interactively. 

@tpike3 The second part of my original also referred to why you created a new file `docs/tutorials/intro_tutorial_files/intro_tutorial.rst`. I guess this was done by accident?

**Comment by tpike3** [comment]
@Corvince my understanding is to make the following updates to the pull request. 

1. Link in the notebook, concur having an interactive notebook would be useful. I also did not know that the notebook existed until I started making updates. A .ipynb with the code would be very useful I am sure for folks going through the tutorial.

2. Delete the redundant .rst

I will try and have it done by next week.

**Comment by jackiekazil** [comment]
@tpike3 - I changed the branch to doc_builds in order to do a test build of the docs through readthedocs before we continue. 

@Corvince - feel free to merge when you think this is ready OR at least give it a thumbs up and I can do the readthedocs build. I can't remember if you have access or not. Feel free to do it if you do. Once everything looks good on RTD, then we can merge into master directly.

File: docs/tutorials/intro_tutorial.rst (hunk 2 above)
Comment (Corvince) [reply docs/tutorials/intro_tutorial.rst:428]: would be great (otherwise copy and paste wont work)

File: docs/tutorials/intro_tutorial.rst (hunk 3 above)
Comment (Corvince) [reply docs/tutorials/intro_tutorial.rst:444]: Since it is consistent I don't think it is an issue, but if you don't mind could you change it back?

**Comment by Corvince** [comment]
@jackiekazil I just tried to push a RTD build for `doc_builds`, but then realized that it doesn't work this way - Toms changes are of course not yet in the `doc_builds` branch. Is there any way to build the PR directly or do we have to merge this one and if it looks good on RTD merge from `doc_builds` into `master` ?

@tpike3 Forgot to mention earlier: I think you did a very good job with this PR!  My comments are merely nitpicks

**Comment by jackiekazil** [comment]
@Corvince - I am a super expert, but i think the only way to do it is if @tpike3 built it and then shared that. Since this was what I am guessing to a one off, that is what I was offering the doc_builds branch.

**Comment by jackiekazil** [comment]
@tpike3 is becoming a doc maintainer. I am going to merge this into doc-builds branch, so he can build and we can all see it. Another PR will be appearing sometime in the near future (from the doc-builds to master) where we will continue this conversation.

**Pull request merged**